### DISCUSSION TITLE
[RFC] Update axe-core to the latest version 4.8.2

### DISCUSSION THREAD

**Discussion by ankit-dq** [storybookjs/storybook#24587] [RFC]
### Summary

Hi Team,

The storybook addon for a11y is a  very nifty and useful tool for finding accessibility issues.
I noticed that it is currently running on axe-core 4.2.0 .
There have been a number of updates to the axe-core version, with the latest update being 4.8.2 .
This update in axe-core has a lot of bug fixes and has also made considerable improvements and enhancements to the axe-core rulesets.

It would be great if you could upgrade the **storybook-addon-a11y** to run using the latest axe-core version 4.8.2

### Problem Statement

I noticed that [https://www.npmjs.com/package/@storybook/addon-a11y](url)  is currently using axe-core 4.2.0 .
There have been a number of updates to the axe-core version, with the latest update being 4.8.2 .
This update in axe-core has a lot of bug fixes and has also made considerable improvements and enhancements to the axe-core rulesets.

### Implementation

Upgrade the axe-core version to the latest i.e. 4.8.2.

### Unresolved Questions

- [ ] Using a to do list makes it easy to resolve the questions as we move the RFC along.

**Comment by JaiRai0304** [comment]
I too uses [@storybook/addon-a11y](https://github.com/storybookjs/storybook/discussions/url) package to scan for accessibility issues for my components. 

It would be really helpful if the package is updated to use the latest axe-core version of 4.8.2

Thank you!

**Comment by JReinhold** [comment]
I think this is a great idea, the only concern is if we should consider this a breaking change or not?

If upgrading `axe-core` introduces more rules, or renames rules, users would all of a sudden get different reports when they upgrade Storybook, which could be considered breaking.

A counter point is that it's clearly just minor bumps for `axe-core` so _they_ don't consider it breaking changes.

Thoughts @shilman

**Reply by JaiRai0304** [reply]
I don't think it breaks anything. But yes it can definately report additional accessibility issues based on the latest ruleset of `axe-core.` In my opinion, Users would love to have  their contents to be tested against the latest ruleset.

**Reply by shilman** [reply]
Yeah that sounds like a breaking change. Good thing 8.0 is on the way!

**Comment by ndelangen** [comment]
The dependency we have on `axe-core` is here:
https://github.com/storybookjs/storybook/blob/8466cf94f7334d0b5ee52a7daed7e3805c67452e/code/addons/a11y/package.json#L56

It's version range is prefixed with `^` meaning users' should be able to use it with version `4.8.x` already.
In fact I'd assume most users would already be doing so.

Maybe I'm missing it, but what is the change actually being requested here?

**Comment by shilman** [comment]
Coming back to this, I agree with @ndelangen 's take above. Installing `addon-a11y` in a new project will get the latest version of `axe-core`. And if you've already installed `addon-a11y` you can just upgrade with your package manager. Closing this discussion.
